dev/generator.jl

```diff
@@ -0,0 +1,35 @@
+using Pkg
+Pkg.activate(@__DIR__)
+Pkg.instantiate()
+
+using Artifacts
+tarball = artifact"P4est"
+cp(joinpath(artifact"P4est", "include"), "p4est_include"; force = true)
+
+using Clang.Generators
+# using Clang.LibClang.Clang_jll  # replace this with your jll package
+
+cd(@__DIR__)
+
+include_dir = joinpath(@__DIR__, "p4est_include")
+
+options = load_options(joinpath(@__DIR__, "generator.toml"))
+
+# add compiler flags, e.g. "-DXXXXXXXXX"
+args = get_default_args()  # Note you must call this function firstly and then append your own flags
+push!(args, "-I$include_dir")
+# push!(args, "-I$include_dir", "-I/usr/lib/x86_64-linux-gnu/openmpi/include")
```

Comment:
You may want to check 

https://github.com/JuliaParallel/MPI.jl/blob/4b9b05f482fb24eab1db2f32352626004b9ebb1c/gen/src/MPIgenerator.jl#L27-L32

for solving the missing `mpi.h` errors.

`push!(args, "-isyetem=$mpi_include_dir")`

Reply:
Thanks!